### PR TITLE
Add cross-entropy loss for semantic segmentation

### PR DESCRIPTION
**Pull request by stegmuel**
## What has changed and why?

Added pixel-wise cross entropy loss for semantic segmentation.

## How has it been tested?

Dummy run using the linear segmentation model's output and dummy masks.
Proper tests will follow.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

File: src/lightly_train/_task_models/dinov2_semantic_segmentation/dinov2_semantic_segmentation_ce_loss.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        Args:
+            logits: Tensor of shape (B, num_classes, H, W)
+                with raw model logits (before softmax).
+            targets: Tensor of shape (B, H, W) with class indices
+                in the range [0, num_classes - 1].
+
+        Returns:
+            Scalar tensor representing the pixel-wise cross-entropy loss.
+        """
+        return F.cross_entropy(
```
Comment (liopeer) [inline src/lightly_train/_task_models/dinov2_semantic_segmentation/dinov2_semantic_segmentation_ce_loss.py:59]: I could possibly see us to use input assertions here to make sure that the right input shapes arrive here at the loss funciton.

File: src/lightly_train/_task_models/dinov2_semantic_segmentation/dinov2_semantic_segmentation_ce_loss.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+class DINOv2SemanticSegmentationCrossEntropyLoss(Module):
+    """
+    Computes pixel-wise cross-entropy loss for semantic segmentation.
+
+    Expects model logits of shape (B, num_classes, H, W) and ground-truth
+    masks of shape (B, H, W) containing integer class indices.
+
+    The loss ignores the specified ignore_index pixels.
```
Comment (liopeer) [inline src/lightly_train/_task_models/dinov2_semantic_segmentation/dinov2_semantic_segmentation_ce_loss.py]: I would be a bit more specific here: AFAIK it ignores the pixel that belong to classes in the `ignore_index`, right?

Comment (stegmuel) [reply src/lightly_train/_task_models/dinov2_semantic_segmentation/dinov2_semantic_segmentation_ce_loss.py]: Yes.
